google/auth/_default_async.py

```diff
@@ -36,10 +35,7 @@ def _warn_about_problematic_credentials(credentials):
     are problematic because they may not have APIs enabled and have limited
     quota. If this is the case, warn about it.
     """
-    from google.auth import _cloud_sdk
-
-    if credentials.client_id == _cloud_sdk.CLOUD_SDK_CLIENT_ID:
-        warnings.warn(_default._CLOUD_SDK_CREDENTIALS_WARNING)
+    return _default._warn_about_problematic_credentials(credentials)
```

Comment:
99% sure you can just import _warn_about_problematic credentials and not wrap it here, further reducing duplication.

Reply:
Makes sense, made the modification to directly import and remove this method.